src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -149,7 +150,8 @@
          {:color (colors/theme-colors colors/neutral-100 colors/white)}]]))])
 
 (defn view
-  [{:keys [on-start-recording on-send on-cancel on-reviewing-audio record-audio-permission-granted
+  [{:keys [on-init on-start-recording on-send on-cancel on-reviewing-audio
```

Comment:
**<< Just opinion >>**

@briansztamfater, my gripe with this function is that it's a monster. It's probably the longest function in the entire codebase, with almost 400 lines of dense packed Clojure. I know, it's complex, lots of things going on, but I cannot imagine how this function can be used in a REPL-driven fashion, with smaller functions that can be called individually. I think this god function deserves a breakdown in a separate PR, if priorities allow and you agree of course. Personally, if I had to modify this file this would be one of the first things I would do to try to get a better REPL experience.

Reply:
@briansztamfater I am wondering why react-native-gesture-handler was not used to handle these gestures.

Reply:
@OmarBasem @ilmotta Honestly I didn't feel it necessary given the requirements of the animation. Most parts are just detecting start of the touch, and if the touch is accepted then track its position, relatively simple stuff and the native API covered the requirements -- animations are then triggered using reanimated. I am not sure how much simpler would have been the implementation using react-native-gesture-handler, but I agree this function it's worth a refactor, I'll create an issue so we can track it.

Reply:
I believe RNGH v2 API is actually simpler using the new `gesture-detector`. And it would be good to unify our usage of APIs. Anyways at least a refactor should do for now.